edge-util/src/Microsoft.Azure.Devices.Edge.Util/metrics/prometheus.net/MetricsProvider.cs

```diff
@@ -13,16 +13,20 @@ public class MetricsProvider : IMetricsProvider
     {
         const string CounterNameFormat = "{0}_{1}_total";
         const string NameFormat = "{0}_{1}";
-        const string InstanceFile = "metrics_instance";
+        const string InstanceFileName = "metrics_instance";
         readonly string namePrefix;
         readonly List<string> defaultLabelNames;
-        static readonly Lazy<string> instanceNumber = new Lazy<string>(GetInstanceNumber);
+        // static readonly Lazy<string> instanceNumber = new Lazy<string>(GetInstanceNumber);
 
-        public MetricsProvider(string namePrefix, string iotHubName, string deviceId)
+        public MetricsProvider(string namePrefix, string iotHubName, string deviceId, string storagePath)
         {
             this.namePrefix = Preconditions.CheckNonWhiteSpace(namePrefix, nameof(namePrefix));
+
+
             Preconditions.CheckNonWhiteSpace(iotHubName, nameof(iotHubName));
             Preconditions.CheckNonWhiteSpace(deviceId, nameof(deviceId));
+            Preconditions.CheckNonWhiteSpace(storagePath, nameof(storagePath));
+            Lazy<string> instanceNumber = new Lazy<string>(() => GetInstanceNumber(storagePath));
```

Comment:
I preserved the pattern of being lazy, but I don't think it needs to be. Can I simply return a string? I'm assuming I am missing the point here and Lazy was intentional.